cmd/thanos/tools_bucket.go

```diff
@@ -710,3 +711,44 @@ func compare(s1, s2 string) bool {
 	}
 	return s1Time.Before(s2Time)
 }
+
+func registerBucketMarkBlock(app extkingpin.AppClause, objStoreConfig *extflag.PathOrContent) {
+	cmd := app.Command(component.Mark.String(), "Mark block for deletion or no-compact in a safe way. NOTE: If the compactor is currently running compacting same block, this operation would be potentially a noop.")
+	blockIDs := cmd.Flag("id", "ID (ULID) of the blocks to be marked for deletion (repeated flag)").Required().Strings()
+	// TODO(bwplotka): Add no-compact option once https://github.com/thanos-io/thanos/pull/3409 is merged.
+	// marker := cmd.Flag("marker", "").Required().Enum(metadata.DeletionMarkFilename)
```

Comment:
Do we want to keep this commented code?

Reply:
Or it is related to the TODO?

Reply:
We could wait with this PR for https://github.com/thanos-io/thanos/pull/3409 so I can uncomment :hugs: